src/client/components/vision2/camera/model.ts

```diff
@@ -79,6 +79,21 @@ export class CameraModel {
       ...opts,
     })
   }
+
+  copy(that: CameraModel) {
+    this.id = that.id
+    this.name = that.name
+    this.image = that.image
+    this.params.copy(that.params)
+    if (this.greenhorizon && that.greenhorizon) {
+      this.greenhorizon.copy(that.greenhorizon)
+    } else {
+      this.greenhorizon = that.greenhorizon
+    }
+    this.balls = that.balls // TODO: deep copy?
+    this.goals = that.goals // TODO: deep copy?
```

Comment:
I don't know... should you deep copy?

Reply:
Maybe! There are potential performance wins for reusing meshes. Currently this blows them all away.

Not critical either way as the expensive shaders are still cached.